recipe/reader.go

```diff
@@ -94,3 +102,10 @@ func (r *Reader) readDir(path string) (recipes []Recipe, err error) {
 
 	return
 }
+
+func validateRecipeVersion(receivedVersion, expectedVersion string) (err error) {
+	if strings.Compare(receivedVersion, expectedVersion) == 0 {
+		return
+	}
+	return fmt.Errorf("received recipe version %s does not match to the expected version %s", receivedVersion, expectedVersion)
+}
```

Comment:
So, shall we design this method to match with all the versions in the `recipeVersions` array, instead of just matching with the last element? Then in the future, we will just have to add versions to the `recipeVersions` array.

Reply:
So, I think what we need here should be that `v1beta1` will be compatible with `v1beta2` up till `v1` right, and we can throw a warning about the migration to a newer version. And then if there are breaking changes `v2---` comes in and only then we throw an error??

Reply:
The idea of matching it with the last element is just to see if it is following the latest recipe version.

And in case there are breaking changes and newer versions, we had a discussion of providing proper alternatives such as migration or version mapping to users.